modules/memberships/class-jetpack-memberships.php

```diff
@@ -277,5 +288,58 @@ public static function get_blog_id() {
 	public static function get_connected_account_id() {
 		return get_option( self::$connected_account_id_option_name );
 	}
+
+	/**
+	 * Whether Recurring Payments are enabled.
+	 *
+	 * @return bool
+	 */
+	public static function is_enabled_jetpack_recurring_payments() {
+		// For WPCOM sites.
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM && function_exists( 'has_any_blog_stickers' ) ) {
+			$site_id = get_current_blog_id();
+			return has_any_blog_stickers( array( 'personal-plan', 'premium-plan', 'business-plan', 'ecommerce-plan' ), $site_id );
+		}
+
+		// For Jetpack sites.
+
+		// FIXME: We enable Recurring Payments on all Jetpack sites, including those on a Free plan.
+		// The reason is that in Jetpack, we cannot yet support the 'new', Upgrade and Stripe Nudge-based
+		// flow, since the post-checkout redirect skips the Jetpack Onboarding Checklist in Calypso,
+		// where Akismet and VaultPress installation is triggered.
+		// Instead, we continue to show the legacy, in-block prompts to upgrade the site's plan, and to
+		// connect Stripe. This however means that we have to display the block also if the site's on a Free plan.
+		//
+		// Check p7rd6c-23C-p2 for details and progress. If ready, uncomment the right-hand side of the line with the
+		// return statement below.
+		// phpcs:ignore
+		return Jetpack::is_active() && (
+			! apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false ) || // Remove when the default becomes `true`
```

Comment:
Would be good to add a comment pointing here in `Jetpack_Gutenberg` at the canonical location of this filter:

https://github.com/Automattic/jetpack/blob/21347595efacb02335f24bf3af236573c2211fef/class.jetpack-gutenberg.php#L202

Reply:
Good spot; I'll add the standard `/** This filter is documented in class.jetpack-gutenberg.php */` banner

Reply:
added in https://github.com/Automattic/jetpack/pull/13394/commits/8ba084f826ea78bd30bc900c76af7a1215c2a13f

Reply:
I really meant that that filter _there_ should have a note to remove this line when the default value is updated.

Reply:
Gotcha 👍  added https://github.com/Automattic/jetpack/pull/13394/commits/19e8af690877a8d864047a04e89383ead61c1384